lib/puma/server.rb

```diff
@@ -131,7 +133,7 @@ def inherit_binder(bind)
     class << self
       # @!attribute [r] current
       def current
-        Thread.current[THREAD_LOCAL_KEY]
+        Thread.current.puma_server
```

Comment:
I'm curious - what's the reason it can't be `Thread.current.thread_variable_get` and `Thread.current.thread_variable_set` instead? They operate exclusively at the thread level "and do not respect fibers" https://ruby-doc.org/core-2.6.3/Thread.html#method-i-thread_variable_get

Reply:
Both approaches support all currently tested versions of Ruby.

As discussed above, `Thread.current.puma_server` is more efficient.